scipy/signal/signaltools.py

```diff
@@ -163,28 +162,28 @@ def correlate(in1, in2, mode='full'):
     elif not in1.ndim == in2.ndim:
         raise ValueError("in1 and in2 should have the same dimensionality")
 
-    if mode == 'valid':
-        _check_valid_mode_shapes(in1.shape, in2.shape)
-        # numpy is significantly faster for 1d
-        if in1.ndim == 1 and in2.ndim == 1:
-            return np.correlate(in1, in2, mode)
+    # numpy is significantly faster for 1d
+    if in1.ndim == 1 and in2.ndim == 1 and (in1.size >= in2.size):
```

Comment:
and for the record, it's because `same` mode in scipy uses the size of the first input, while `same` mode in numpy uses the size of the larger input

(so numpy's could be used for *any* 1D arrays with some before and after swapping?  I'm not sure)